### ISSUE TITLE
Are the `.xacro` and `_macro.xacro` copies of `ros-industrial/abb_experimental`?

### ISSUE THREAD

**Issue by gavanderhoorn** [alexarbogast/abb_irb4600_description#1]
Hi. Happened to come across your repository.

Nice work on the meshes, but it appears [abb_irb4600_40_255_macro.xacro](https://github.com/alexarbogast/abb_irb4600_description/blob/184db58d61c3fec1f625c4ccf1ca82848eaa92d1/urdf/abb_irb4600_40_255_macro.xacro) and [abb_irb4600_40_255.xacro](https://github.com/alexarbogast/abb_irb4600_description/blob/184db58d61c3fec1f625c4ccf1ca82848eaa92d1/urdf/abb_irb4600_40_255.xacro) are almost direct copies from the same files in [ros-industrial/abb_experimental](https://github.com/ros-industrial/abb_experimental).

That's OK, as they are licensed Apache 2.0 ([here](https://github.com/ros-industrial/abb_experimental/blob/20b9cf86ad274e4f98540b9c9efe59f94245d2d1/abb_irb4600_support/package.xml#L30)), but that comes with some requirements.

It's mentioned in the license itself (section 4), but to summarise: you can't copy files or content of files without attribution and you can't always relicense files.

See [Open Source Licenses 101: Apache License 2.0](https://fossa.com/blog/open-source-licenses-101-apache-license-2-0) for a (very terse) overview.

Could you please take care of this and comply with the `Apache-2.0` license terms?

**Comment by alexarbogast** [comment]
Thank you for bringing this to my attention, and thanks for all you do for the ROS community! 

I am fairly new to this, so I hope the changes made in af4506fd7ab19d18b015ce00b737c21ca937bb7b are satisfactory for `Apache-2.0` compliance. I see that section 4 states,
> You must cause any modified files to carry prominent notices stating that You changed the files

I read [here](https://opensource.stackexchange.com/a/13150) that appending a personal copyright line in file headers is sufficient for the modification notice. However, the original xacro files do not contain the Apache License header. 

I also have an upstream repo and ROS metapackage [alexarbogast/abb_irb4600_ros](https://github.com/alexarbogast/abb_irb4600_ros.git) that includes this robot description as a submodule. Should this repo also include the license?

Please let me know if there are any further changes that should be made. Thanks again!

**Comment by gavanderhoorn** [comment]
tl;dr: it isn't that different from academic references. Keep/add attribution (ie: mention original authors/maintainers), mention where you found something and at which version.

---

Hi, thanks for responding.

> I am fairly new to this, so I hope the changes made in [af4506f](https://github.com/alexarbogast/abb_irb4600_description/commit/af4506fd7ab19d18b015ce00b737c21ca937bb7b) are satisfactory for `Apache-2.0` compliance. 

So while adding the license file is a good step -- note: there wasn't necessarily a requirement to change your license to Apache 2.0 -- it's not sufficient.

Whenever you copy (a) file(s), you'll have to make sure to mention their provenance and keep copyright notices and attribution in place as well.

In your case, that means stating somewhere -- clearly visible -- you copied those two `.xacro`s from `ros-industrial/abb_experimental` and mention the original authors / contributors.

> I see that section 4 states,
> 
> > You must cause any modified files to carry prominent notices stating that You changed the files
> 
> I read [here](https://opensource.stackexchange.com/a/13150) that appending a personal copyright line in file headers is sufficient for the modification notice. However, the original xacro files do not contain the Apache License header.

You could just add one. That's typically done in these cases.

> I also have an upstream repo and ROS metapackage [alexarbogast/abb_irb4600_ros](https://github.com/alexarbogast/abb_irb4600_ros.git) that includes this robot description as a submodule. Should this repo also include the license?

No, I don't believe so, as this repository is a 'stand-alone project'. It just happens to be a dependency of your other project.

**Comment by alexarbogast** [comment]
Thank you for the detailed response. I have added the license header and a declaration of modifications for both `.xacro`s in 9e71ff79c4d6ae2c7cc2acab868dc0dc6167bd52. The authors and maintainers of the `abb_irb4600_support` package (as declared in the package manifest) have been added to the copyright notice.

Thank you for your help with this, and please let me know if there are any further modifications that should be made.
